### ISSUE TITLE
Bot randomly crashes when pathing.

### ISSUE THREAD

**Issue by cohenerickson** [PrismarineJS/mineflayer-pathfinder#285]
Might be a similar issue to #83
```
"dependencies": {
    "minecraft-data": "^3.9.1",
    "mineflayer": "^4.3.0",
    "mineflayer-pathfinder": "^2.3.0",
    "vec3": "^0.1.7"
  },
```
```
C:\Users\cohen\Desktop\mineflayer\experimenting\node_modules\mineflayer-pathfinder\lib\movements.js:212
    const y = pos.y + dy
                  ^
TypeError: Cannot read properties of undefined (reading 'y')
    at Movements.getNumEntitiesAt (C:\Users\cohen\Desktop\mineflayer\experimenting\node_modules\mineflayer-pathfinder\lib\movements.js:212:19)
    at Movements.safeOrBreak (C:\Users\cohen\Desktop\mineflayer\experimenting\node_modules\mineflayer-pathfinder\lib\movements.js:286:18)
    at Movements.getMoveDiagonal (C:\Users\cohen\Desktop\mineflayer\experimenting\node_modules\mineflayer-pathfinder\lib\movements.js:427:19)
    at Movements.getNeighbors (C:\Users\cohen\Desktop\mineflayer\experimenting\node_modules\mineflayer-pathfinder\lib\movements.js:654:12)
    at AStar.compute (C:\Users\cohen\Desktop\mineflayer\experimenting\node_modules\mineflayer-pathfinder\lib\astar.js:85:40)
    at EventEmitter.monitorMovement (C:\Users\cohen\Desktop\mineflayer\experimenting\node_modules\mineflayer-pathfinder\index.js:388:36)
    at EventEmitter.emit (node:events:527:28)
    at EventEmitter.emit (node:domain:475:12)
    at Timeout.doPhysics [as _onTimeout] (C:\Users\cohen\Desktop\mineflayer\experimenting\node_modules\mineflayer\lib\plugins\physics.js:66:13)
    at listOnTimeout (node:internal/timers:559:17)
    ```

**Comment by AmarUCLA** [comment]
Having the same issue. It seems like when the bot is pathing it would crash randomly because of this same error.

**Comment by oscarspalk** [comment]
All of the times of which I have encountered this error, it was either because the goal was too far away or because I had used the wrong constructor for Vec3.

**Comment by gamerboytr** [comment]
same error

**Comment by oscarspalk** [comment]
can we have a look at the code

**Comment by VakinhaMuMu** [comment]
Im having the same error. The bot starts normally and suddenly crashes when goal is too far

**Comment by oscarspalk** [comment]
@VakinhaMuMu which is to be expected. You can't get a property of a block of which you haven't loaded yet.

**Comment by cohenerickson** [comment]
It seemed to happen somewhat randomly, here is the basis of what my code did.

create an XZ goal with cords 1000, and 1000, and start the bot pathing.

after a hundred blocks or so the bot would suddenly log offline with this error in the console.

I don't think it is an issue with unloaded chunks because it had pathed quite a distance before crashing on some occasions.

**Comment by LionTion** [comment]
I had this problem too and fixed it by adding one line to the function in movements.js

Step1: Find the Module either by searching or going to the directory directly if you know where it is
Step 2: Go under the folder lib to movements.js
Step 3: Find "getNumEntitiesAt" and the first result should be a function at Line 210
Step 4: Add the following Line under the first instruction
`if (pos == null) return 0`
Step 5: Test if it still crashes

The function should look like this by now:
``` 
getNumEntitiesAt (pos, dx, dy, dz) {
    if (this.allowEntityDetection === false) return 0
    if (pos == null) return 0
    const y = pos.y + dy
    const x = pos.x + dx
    const z = pos.z + dz

    return this.entityIntersections[`${x},${y},${z}`] ?? 0
  }
```

Might not be a good fix but it works for me and hopefully for others too

**Comment by eihqnh** [comment]
same error

**Comment by IceTank** [comment]
This should be fixed in version 2.3.1. If there are any more issues re open this issue or make a new issue. To upgrade to the newest version off mineflayer-pathfinder you can use `npm update`.
